sig/prism/desugar_compiler.rbs

```diff
@@ -0,0 +1,49 @@
+module Prism
+  class DesugarCompiler < MutationCompiler
```

Comment:
I'm not sure I actually want this entire compiler documented. The only thing you should know is that you do:

```ruby
node.accept(DesugarCompiler.new)
```

you shouldn't know about the inner workings of it. Especially not the private functions. Is there a way we can hide the implementation details here?